receiver/dotnetdiagnosticsreceiver/config.go

```diff
@@ -0,0 +1,40 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dotnetdiagnosticsreceiver
+
+import "go.opentelemetry.io/collector/config/configmodels"
+
+type Config struct {
+	configmodels.ReceiverSettings `mapstructure:",squash"`
+	// Optional directory name to save binary blobs read from the socket. This can
+	// be useful for troubleshooting/debugging. If left blank or unset, no blobs are
+	// saved.
+	BlobDir string `mapstructure:"blob_dir"`
+	// The maximum number of blob files to keep in the blob_dir, if specified. Defaults
+	// to 10 (most recent).
+	MaxBlobFiles int `mapstructure:"max_blob_files"`
+	// The process ID of the dotnet process from which to collect diagnostics.
+	PID int `mapstructure:"pid"`
```

Comment:
It's not captured dynamically at this level, so you do have to put in the process ID. I'm not sure yet how to best make process discovery work, because you can get a list of the dotnet processes, but their process names are all just 'dotnet' so I don't see how to distinguish them. Also, if we want dynamic functionality, would that be better suited to using an observer with a receiver creator? Let's discuss separately.

Reply:
> if we want dynamic functionality, would that be better suited to using an observer with a receiver creator? Let's discuss separately.

Yes, let's separate this discussion. Right now I see the need for an observer.